pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -484,21 +547,136 @@ func serviceAccountNameToPodSelector(saName string) *metav1.LabelSelector {
 func hasPerNamespaceRule(cnp *crdv1alpha1.ClusterNetworkPolicy) bool {
 	for _, ingress := range cnp.Spec.Ingress {
 		for _, peer := range ingress.From {
-			if peer.Namespaces != nil && peer.Namespaces.Match == crdv1alpha1.NamespaceMatchSelf {
+			if peer.Namespaces != nil {
 				return true
 			}
 		}
 	}
 	for _, egress := range cnp.Spec.Egress {
 		for _, peer := range egress.To {
-			if peer.Namespaces != nil && peer.Namespaces.Match == crdv1alpha1.NamespaceMatchSelf {
+			if peer.Namespaces != nil {
 				return true
 			}
 		}
 	}
 	return false
 }
 
+func namespaceRuleLabelKeys(cnp *crdv1alpha1.ClusterNetworkPolicy) sets.Set[string] {
+	keys := sets.New[string]()
+	for _, ingress := range cnp.Spec.Ingress {
+		for _, peer := range ingress.From {
+			if peer.Namespaces != nil && len(peer.Namespaces.SameLabels) > 0 {
```

Comment:
No need to check `len(peer.Namespaces.SameLabels)>0`. Same reason as you didn't check `Spec.Ingress` or `ingress.From` length > 0.